test/stdlib/Dispatch.swift

```diff
@@ -1,6 +1,7 @@
 // RUN: %target-run-simple-swift
 // REQUIRES: executable_test
 // REQUIRES: libdispatch
+// XFAIL: OS=linux-gnu
```

Comment:
```
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:14:42: error: type 'DispatchSource' has no member 'ProcessEvent'
  expectEqual(2147483648, DispatchSource.ProcessEvent.exit.rawValue)
                          ~~~~~~~~~~~~~~ ^~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:87:13: error: use of unresolved identifier 'mach_timebase_info_data_t'
        var info = mach_timebase_info_data_t(numer: 1, denom: 1)
                   ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:88:2: error: use of unresolved identifier 'mach_timebase_info'
        mach_timebase_info(&info)
        ^~~~~~~~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:95:46: error: use of unresolved identifier 'NSEC_PER_SEC'
        time = DispatchTime(uptimeNanoseconds: 15 * NSEC_PER_SEC)
                                                    ^~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:96:43: error: use of unresolved identifier 'NSEC_PER_SEC'
        expectEqual(time.uptimeNanoseconds, 15 * NSEC_PER_SEC)
                                                 ^~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:148:14: error: use of unresolved identifier 'NSEC_PER_SEC'
        expectEqual(NSEC_PER_SEC, diffNanos)
                    ^~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:153:66: error: use of unresolved identifier 'O_RDONLY'
        let chan = DispatchIO(type: .random, path: "_REL_PATH_", oflag: O_RDONLY, mode: 0, queue: q, cleanupHandler: { (error) in })
                                                                        ^~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:200:36: error: argument type 'DispatchTimeInterval' does not conform to expected type 'Equatable'
                expectEqual(DispatchTimeInterval.seconds(i), DispatchTimeInterval.milliseconds(i * 1000))
                                                 ^
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:201:36: error: argument type 'DispatchTimeInterval' does not conform to expected type 'Equatable'
                expectEqual(DispatchTimeInterval.milliseconds(i), DispatchTimeInterval.microseconds(i * 1000))
                                                 ^
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:202:36: error: argument type 'DispatchTimeInterval' does not conform to expected type 'Equatable'
                expectEqual(DispatchTimeInterval.microseconds(i), DispatchTimeInterval.nanoseconds(i * 1000))
                                                 ^
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:228:46: error: binary operator '!=' cannot be applied to two 'DispatchTimeInterval' operands
        expectTrue(DispatchTimeInterval.seconds(10) != DispatchTimeInterval.never);
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:228:46: note: binary operator '!=' cannot be synthesized for enums with associated values
        expectTrue(DispatchTimeInterval.seconds(10) != DispatchTimeInterval.never);
                                                    ^
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:229:40: error: binary operator '!=' cannot be applied to two 'DispatchTimeInterval' operands
        expectTrue(DispatchTimeInterval.never != DispatchTimeInterval.seconds(10));
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/yln/work/swift-source/swift/test/stdlib/Dispatch.swift:229:40: note: binary operator '!=' cannot be synthesized for enums with associated values
        expectTrue(DispatchTimeInterval.never != DispatchTimeInterval.seconds(10));
                                              ^
```

Reply:
Sounds like `-DDEPLOYMENT_TARGET_LINUX` may be missing?

Reply:
I searched for `DEPLOYMENT_TARGET_LINUX` and it seems to be a swift-corelibs-foundation option. Currently I am building simply via `./swift/utils/build-script  --release-debuginfo  --libdispatch --install-libdispatch`, e.g. only worrying about libdispatch at the moment.

Do you have an idea how to make `import Dispatch` work in the tests without copying things into the <swift-build> directory?

Reply:
Add search paths using lit, I think.

Reply:
For the link, you will need to add `-L` for the search paths.  For the runtime, you will need to add `LD_LIBRARY_PATH` on Linux, `DYLD_LIBRARY_PATH` on macOS, and `PATH` on Windows.